pkg/operator/starter.go

```diff
@@ -50,62 +52,95 @@ const (
 	infrastructureName = "cluster"
 )
 
-func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext) error {
-	// Create core clientset and informer
-	kubeClient := kubeclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, operatorName))
-	kubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(kubeClient, defaultNamespace, cloudConfigNamespace, "")
-	secretInformer := kubeInformersForNamespaces.InformersFor(defaultNamespace).Core().V1().Secrets()
-	nodeInformer := kubeInformersForNamespaces.InformersFor("").Core().V1().Nodes()
-	configMapInformer := kubeInformersForNamespaces.InformersFor(defaultNamespace).Core().V1().ConfigMaps()
-
-	// Create config clientset and informer. This is used to get the cluster ID
-	configClient := configclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, operatorName))
-	configInformers := configinformers.NewSharedInformerFactory(configClient, 20*time.Minute)
-	infraInformer := configInformers.Config().V1().Infrastructures()
-
-	// Create informer for the ConfigMaps in the openshift-config-managed namespace. This is used to get the custom CA
-	// bundle to use when accessing the AWS API.
-	cloudConfigInformer := kubeInformersForNamespaces.InformersFor(cloudConfigNamespace).Core().V1().ConfigMaps()
-	cloudConfigLister := cloudConfigInformer.Lister().ConfigMaps(cloudConfigNamespace)
-
-	// Create GenericOperatorclient. This is used by the library-go controllers created down below
-	gvr := opv1.SchemeGroupVersion.WithResource("clustercsidrivers")
-	operatorClient, dynamicInformers, err := goc.NewClusterScopedOperatorClientWithConfigName(controllerConfig.KubeConfig, gvr, instanceName)
+func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext, guestKubeConfigString *string) error {
+	// Create core clientset and informer for the MANAGEMENT cluster.
+	eventRecorder := controllerConfig.EventRecorder
+	controlPlaneNamespace := controllerConfig.OperatorNamespace
+	controlPlaneKubeClient := kubeclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, operatorName))
+	controlPlaneKubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(controlPlaneKubeClient, controlPlaneNamespace, cloudConfigNamespace, "")
```

Comment:
I don't think it's correct for the operator in a mgmt cluster to read config maps in another namespace in HyperShift.

Reply:
This was here because I was wrongly creating SCs in the management cluster. Should be OK to remove now.

Reply:
I can still see `cloudConfigNamespace` for CustomAWSBundleSyncer that's used only in the standalone OCP. Will `NewKubeInformersForNamespaces(.., cloudConfigNamespace)`  watch anything in the namespace, even when there is no Informer needed there? I hope it won't, because I don't want to give the operator RBAC to read cloudConfigNamespace in the mgmt cluster.